src/testing/__tests__/recording.test.ts

```diff
@@ -51,6 +51,38 @@ test('create files relative to the recording directory based on the input name',
   ).toEqual(true);
 });
 
+test('accepts recordFailedRequests PollyConfig option', async () => {
+  // start new server which responds with statusCode=404
+  server.close();
+  server = await startServer(404);
+
+  const name = 'test-record-failed-requests';
+  recording = setupRecording({
+    name,
+    directory: __dirname,
+    options: {
+      recordFailedRequests: true,
+    },
+  });
+
+  await fetch(`http://localhost:${server.port}`);
+
+  await recording.stop();
+
+  expect(Object.keys(vol.toJSON())).toHaveLength(1);
+
+  const [recordingPath] = Object.keys(vol.toJSON());
+
+  expect(
+    recordingPath.startsWith(
+      toUnixPath(path.resolve(__dirname, '__recordings__', name)),
+    ),
+  ).toEqual(true);
+
+  server.close();
+  server = await startServer();
```

Comment:
This will fail to reset `server` if the `expect` fails I think.

Reply:
I feel like it'd be good to just restart the server in `beforeEach` and `afterEach` now that it's being configured differently in different tests... unless @charlieduong94 would prefer to leave them in `beforeAll`/`afterAll`.

Reply:
Yep, using `beforeEach`/`afterEach` would be better now.